### PR TITLE
Archive/delete process manager upon certain events or rejections

### PR DESCRIPTION
**Pull request by dmitrykuzmin**
This PR addresses #920.

It adds the configurable lifecycle rules to the `ProcessManagerRepository`.

The rules can be configured as follows:
```java
repository.lifecycle()
          .archiveOn(Event1.class, Rejection1.class)
          .deleteOn(Rejection2.class); 
```
Now, all managed entities will be automatically archived upon emitting `Event1` or `Rejection1` and will be automatically deleted on producing `Rejection2`.

The archivation/deletion takes place in scope of the same transaction where events were emitted.

### CODE REVIEW

**Comment by dmitrykuzmin** [comment]
@armiol PTAL. The build will pass once https://github.com/SpineEventEngine/base/pull/339 is merged.

**Comment by dmitrykuzmin** [comment]
@armiol PTAL again.

File: core/src/main/java/io/spine/core/Events.java [hunk 1]
```diff
@@ -257,6 +258,16 @@ public static RejectionEventContext rejectionContext(CommandMessage commandMessa
                                     .build();
     }
 
+    /**
+     * Obtains the type URL of the event message enclosed by a given {@code Event}.
+     */
+    public static TypeUrl typeUrl(Event event) {
```
Comment (armiol) [inline core/src/main/java/io/spine/core/Events.java]: This is weird. I think it belongs to a message interface, such as `EventMessage`.

Also I suspect this isn't the first time we want to find out the `TypeUrl` of an `Event`. So please have a look around, maybe this task is already solved somehow.

File: server/src/main/java/io/spine/server/procman/Lifecycle.java [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+     */
+    @SafeVarargs
+    @CanIgnoreReturnValue
+    public final Lifecycle archiveOn(Class<? extends EventMessage>... messageClasses) {
+        archiveOn.addAll(asList(messageClasses));
+        return this;
+    }
+
+    /**
+     * Checks if the process hould become archived when the given {@code events} are emitted.
```
Comment (armiol) [inline server/src/main/java/io/spine/server/procman/Lifecycle.java]: `should`.

**Comment by dmitrykuzmin** [comment]
@armiol PTAL.

**Pull request merged**